src/serializer/serializer.js

```diff
@@ -119,12 +120,45 @@ export class Serializer {
     if (timingStats !== undefined) timingStats.deepTraversalTime = Date.now();
     let residualHeapVisitor = new ResidualHeapVisitor(this.realm, this.logger, this.modules);
     residualHeapVisitor.visitRoots();
+    this.realm.wrapInGlobalEnv(() => this.realm.evaluateForEffects(() => {
+      if (this.options.additionalFunctions) {
+        for (let [functionString, effects] of this.functions.writeEffects.entries()) {
+          let [r, g, ob, pb, co] = effects;
+          // need to switch the application order of generators to ensure consistency
+          // between visitor and serializer (a function should have only one associated
+          // generator).
+          let originalLength = g.length;
+          let originalGenerator = this.realm.generator;
+          this.realm.generator = g;
```

Comment:
Ok, looks like I don't actually have to change the `realm.generator` just `this.generator`.